tests/infer/mcmc/test_nuts.py

```diff
@@ -184,7 +184,9 @@ def model(data):
     assert_equal(samples["p_latent"].mean(0), true_probs, prec=0.02)
 
 
-@pytest.mark.parametrize("jit", [False, mark_jit(True)], ids=jit_idfn)
+@pytest.mark.parametrize("jit", [False, mark_jit(True, marks=[
+    pytest.mark.skipif("CUDA_TEST" in os.environ, reason="FIXME: Too small step size with CUDA on JIT.")
+])], ids=jit_idfn)
```

Comment:
It is tricky to debug the root of this. All I can tell is potential_fn gives wrong derivatives under CUDA+JIT for this type of model. Gamma prior does not cause problem because when I replace `Gamma` prior by `HalfNormal(rate)`, I still get the grad issue. This looks weird to me because this issue does not happen for some other models. I think we need to create an issue to track this problem.

Reply:
Thanks for digging into this!